Initial commit of VIS based Vehicle HAL

arminn opened the pull request:
Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
Initial commit of VIS based Vehicle HAL

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
test: Added Visclient Test

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym commented on common/src/VisClient.cpp:
```diff
@@ -0,0 +1,825 @@
+
+    if (mUri.empty()) {
+        char propValue[PROPERTY_VALUE_MAX] = {};
+        property_get("persist.vis.uri", propValue, "wss://192.168.1.100:8088");
```
Is it expected to be hard coded like this?
Maybe put it as define to top of file with explanation in comments?

rshym commented on common/src/VisClient.cpp:
```diff
@@ -0,0 +1,825 @@
+    std::lock_guard<std::mutex> lock(mLock);
+
+    switch (reinterpret_cast<size_t>(user)) {
+        case 1:
```
As I understand these error codes are defined somewhere outside of project.
If this is the case, we may provide comment about source of these numbers, for possible expansion and verification.

rshym commented on common/src/VisVehicleHal.cpp:
```diff
@@ -0,0 +1,629 @@
+            }
+        }
+    }
+#if 0
```
Do we need to keep disabled code?
If it is intended to be used in some cases, may be to add some comments regarding possible use case.

arminn replied:
In short - yes. We need it for debug/simulation.

arminn committed:
build: Reworked Android.bp

Decouple vehicle services, test, libvisclient(will be
removed soon to separate project).

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
vehicle: Decrease log output by framework

Moving PERF_VEHICLE_SPEED from CONTINUOUS to ON_CHANGE,
for decrease log output like: "CarDrivingState: Property Changed".

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
build: Remove include hardcode

According to Google best practices: LOCAL_EXPORT_C_INCLUDE_DIRS/export_include_dirs
should be used in a project which exporting includes, instead of
patch hardcode.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
description: Added README.md

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>
